### PR TITLE
3/27 consumable Build finished

### PR DESCRIPTION
**Pull request by PHONGNGYUEN1**
The build is now working for the Consumable items and its tests

### CODE REVIEW

File: source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        @BeforeEach
+        void setUp() { // Initialize TestableItem and ItemUsageContext
+            item1 = new TestableItem1("Test1", 1, 10, 1);
+            context = new ItemUsageContext();
+        }
+
+        @Test
+        void testIfNotEmpty() {
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java:43]: Create a test which checks the consumption use - ie don't bother overriding the ConsumableItem useItem function (this is what we are trying to test). Create a test object with e.g. 3 quantity, and useItem 3 times and check that quantity is the correct value each time. Then run useItem one more time and assert an error is thrown.

**Review by mitsudz — CHANGES_REQUESTED** [review]
Check comments for changes that need to be made - in particular no need to rewrite functionality provided in AbstractItem - and make sure constructors are written appropriately (I'm not even sure this needs a new constructor). Also make sure tests are actually running and pass. Also please refactor and check over javadocs.

File: source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java (hunk 1 above)
Comment (PHONGNGYUEN1) [reply source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java]: Implemented this into newest commit

File: source/core/src/main/com/csse3200/game/inventory/items/ConsumableItem.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     *  Updates the no. of uses left for each usage
+     *
+     * @param inputs the inputs for this ite to be used
+     */
+    @Override
+    public void useItem(ItemUsageContext inputs) {
+        if (super.quantity > 0) {
```
[the remark this answers is not in the document]
Comment (PHONGNGYUEN1) [reply source/core/src/main/com/csse3200/game/inventory/items/ConsumableItem.java]: implemented in newedt build

File: source/core/src/main/com/csse3200/game/inventory/items/ConsumableItem.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * Defines the useItemm function in {@link ItemUsageContext}
+ * Should be input to {@code useItem}.
+ * </p>
+ */
+public abstract class ConsumableItem extends AbstractItem {
+
+    /**
+     * Constructs a ConsumableItem
+     */
+    public ConsumableItem(String name, int itemCode, int limit, int quantity) {
```
Comment (s-shalomi) [inline source/core/src/main/com/csse3200/game/inventory/items/ConsumableItem.java:16]: Try make javadocs more specific

File: source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            else {
+                this.quantity--;
+                super.isEmpty();
+            }
+        }
+    }
+
+    @BeforeEach
+    void setUp() { // Initialize TestableItem and ItemUsageContext
+        item1 = new TestableItem1("Test1", 3, 10, 3);
```
Comment (s-shalomi) [inline source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java:46]: create a new ConsumableItem to test

**Review by s-shalomi — COMMENTED** [review]
Minor changes, but looks good!

File: source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java (hunk 4 above)
Comment (mitsudz) [reply source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java:46]: You cannot create a ConsumableItem - this is an abstract class.

File: source/core/src/main/com/csse3200/game/inventory/items/ConsumableItem.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    /**
+     *  Updates the no. of uses left for each usage
+     *
+     * @param inputs the inputs for this ite to be used
+     */
+    @Override
+    public void useItem(ItemUsageContext inputs) {
+        if (super.isEmpty()) {
+            throw new ArrayIndexOutOfBoundsException("This item has ran out of uses");
```
[the remark this answers is not in the document]
Comment (PHONGNGYUEN1) [reply source/core/src/main/com/csse3200/game/inventory/items/ConsumableItem.java]: would IllegalStateException be better?

**Pull request closed without merging**